src/routes/safe/components/Balances/Collectibles/index.tsx

```diff
@@ -95,46 +97,69 @@ const Collectibles = (): React.ReactElement => {
     setSendNFTsModalOpen(true)
   }
 
-  return (
-    <Card className={classes.cardOuter}>
-      <div className={classes.cardInner}>
-        {/* No collectibles */}
-        {nftAssetsFromNftTokens.length === 0 && (
+  if (nftAssetsFromNftTokens.length === 0) {
+    return (
+      <Card className={classes.cardOuter}>
+        <div className={classes.cardInner}>
           <Paragraph className={classes.noData}>No collectibles available</Paragraph>
-        )}
+        </div>
+      </Card>
+    )
+  }
+
+  const getNftTokens = (index: number) => {
+    const nftAsset = nftAssetsFromNftTokens[index]
+    return nftTokens.filter(({ assetAddress }) => nftAsset.address === assetAddress)
+  }
 
-        {/* collectibles List*/}
-        {nftAssetsFromNftTokens.length > 0 &&
-          nftAssetsFromNftTokens.map((nftAsset) => {
-            return (
-              <Fragment key={nftAsset.slug}>
-                <div className={classes.title}>
-                  <div className={classes.titleImg} style={{ backgroundImage: `url(${nftAsset.image || ''})` }} />
-                  <h2 className={classes.titleText}>{nftAsset.name}</h2>
-                  <div className={classes.titleFiller} />
-                </div>
-                <div className={classes.gridRow}>
-                  {nftTokens
-                    .filter(({ assetAddress }) => nftAsset.address === assetAddress)
-                    .map((nftToken) => (
-                      <Item
-                        data={nftToken}
-                        key={`${nftAsset.slug}_${nftToken.tokenId}`}
-                        onSend={() => handleItemSend(nftToken)}
-                      />
-                    ))}
-                </div>
-              </Fragment>
-            )
-          })}
-      </div>
+  const getItemSize = (index: number) => {
+    const NFT_CARD_HEIGHT = 365
+    const nftAmount = getNftTokens(index).length
+    const rows = Math.ceil(nftAmount / 4)
+    return rows * NFT_CARD_HEIGHT
+  }
+
+  return (
+    <>
+      <AutoSizer>
+        {(dimensions) => {
+          return (
+            <List itemCount={50} itemSize={getItemSize} {...dimensions}>
+              {({ index, style }) => {
+                const nftAsset = nftAssetsFromNftTokens[index]
+                if (!nftAsset) {
```

Comment:
This can be `undefined` when scrolling through the list until it finds the element